app/src/main/java/com/github/se/eduverse/viewmodel/TodoListViewModel.kt

```diff
@@ -0,0 +1,100 @@
+package com.github.se.eduverse.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.github.se.eduverse.model.Todo
+import com.github.se.eduverse.model.TodoStatus
+import com.github.se.eduverse.repository.TodoRepository
+import com.google.firebase.Firebase
+import com.google.firebase.firestore.firestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class TodoListViewModel(private val todoRepository: TodoRepository) : ViewModel() {
+  private val _actualTodos = MutableStateFlow<List<Todo>>(emptyList())
+  val actualTodos: StateFlow<List<Todo>> = _actualTodos.asStateFlow()
+
+  private val _doneTodos = MutableStateFlow<List<Todo>>(emptyList())
+  val doneTodos: StateFlow<List<Todo>> = _doneTodos.asStateFlow()
+
+  // create viewmodel factory
+  companion object {
+    val Factory: ViewModelProvider.Factory =
+        object : ViewModelProvider.Factory {
+          @Suppress("UNCHECKED_CAST")
+          override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            return TodoListViewModel(TodoRepository(Firebase.firestore)) as T
+          }
+        }
+  }
+
+  init {
+    getActualTodos()
+    getDoneTodos()
+  }
+
+  fun getNewUid(): String {
+    return todoRepository.getNewUid()
+  }
+
+  fun getActualTodos() {
+    todoRepository.getActualTodos(
+        onSuccess = { todos -> _actualTodos.value = todos },
+        onFailure = { exception -> Log.e("getActualTodos", "Error getting todos", exception) })
+  }
+
+  fun getDoneTodos() {
+    todoRepository.getDoneTodos(
+        onSuccess = { todos -> _doneTodos.value = todos },
+        onFailure = { exception -> Log.e("getDoneTodos", "Error getting todos", exception) })
+  }
+
+  fun addNewTodo(todo: Todo) {
+    todoRepository.addNewTodo(
+        todo = todo,
+        onSuccess = { getActualTodos() },
+        onFailure = { exception -> Log.e("addNewTodo", "Error adding todo", exception) })
+  }
+
+  /** Sets the todo state to done */
+  fun setTodoDone(todo: Todo) {
+    val doneTodo =
+        Todo(uid = todo.uid, name = todo.name, timeSpent = todo.timeSpent, status = TodoStatus.DONE)
+    todoRepository.updateTodo(
+        todo = doneTodo,
+        onSuccess = {
+          getActualTodos()
+          getDoneTodos()
+        },
+        onFailure = { exception -> Log.e("setTodoDone", "Error updating todo", exception) })
+  }
```

Comment:
You should also move the todo from actualTodo into doneTodo. If you want a todo to stayed displayed when you change it status, then it is alright. **But** if you do so, the buttons "current tasks" and "completed tasks" of the screen should respectively call `getActualTodos` and `getDoneTodos`. This way, the todo that changed of status will be displayed

Reply:
> You should also move the todo from actualTodo into doneTodo. If you want a todo to stayed displayed when you change it status
There's no need to do that since I call getActualTodos and getDoneTodos on success the lists will be updated be these functions, and since I call these functions on success of each function that updates the lists are always kept up to date there's no need for > the buttons "current tasks" and "completed tasks" of the screen should respectively call `getActualTodos` and `getDoneTodos`

Reply:
Oh okay, my mistake then sorry